dev-libs/libusb/patches/libusb-1.0.23.patchset

```diff
@@ -0,0 +1,63 @@
+From 183f5d3ef2e6425427462b78c4725ed679eea57f Mon Sep 17 00:00:00 2001
+From: Ken Mays <kmays2000@gmail.com>
+Date: Thu, 5 Sep 2019 01:15:08 +0000
+Subject: Fix Haiku port
+
+
+diff --git a/libusb/os/haiku_usb_backend.cpp b/libusb/os/haiku_usb_backend.cpp
+index 539a996..16d7528 100644
+--- a/libusb/os/haiku_usb_backend.cpp
++++ b/libusb/os/haiku_usb_backend.cpp
+@@ -309,7 +309,7 @@ USBDeviceHandle::SetAltSetting(int inumber, int alt)
+ 	return LIBUSB_SUCCESS;
+ }
+ 
+-
++/*
+ int
+ USBDevice::ClearHalt(int endpoint)
+ {
+@@ -325,6 +325,23 @@ USBDevice::ClearHalt(int endpoint)
+ 		return _errno_to_libusb(command.control.status);
+ 	}
+ }
++*/
++
++int
++USBDeviceHandle::ClearHalt(int endpoint)
++{
++	usb_raw_command command;
++	command.control.request_type = USB_REQTYPE_ENDPOINT_OUT;
++	command.control.request = USB_REQUEST_CLEAR_FEATURE;
++	command.control.value = USB_FEATURE_ENDPOINT_HALT;
++	command.control.index = endpoint;
++	command.control.length = 0;
++
++	if (ioctl(fRawFD, B_USB_RAW_COMMAND_CONTROL_TRANSFER, &command, sizeof(command)) ||
++			command.control.status != B_USB_RAW_STATUS_SUCCESS) {
++		return _errno_to_libusb(command.config.status);
++	}
++}
+ 
+ 
+ USBDevice::USBDevice(const char *path)
+@@ -449,6 +466,7 @@ USBDevice::EndpointToInterface(uint8 address) const
+ int
+ USBDevice::Initialise()		//Do we need more error checking, etc? How to report?
+ {
++	usbi_dbg("Haiku init dev %s", fPath);
+ 	int fRawFD = open(fPath, O_RDWR | O_CLOEXEC);
+ 	if (fRawFD < 0)
+ 		return B_ERROR;
+@@ -479,7 +497,7 @@ USBDevice::Initialise()		//Do we need more error checking, etc? How to report?
+ 		command.config_etc.descriptor = (usb_configuration_descriptor*)fConfigurationDescriptors[i];
+ 		command.config_etc.length = tmp_config.total_length;
+ 		command.config_etc.config_index = i;
+-		if (ioctl(fRawFD, B_USB_COMMAND_GET_CONFIGURATION_DESCRIPTOR_ETC, &command, sizeof(command)) ||
++		if (ioctl(fRawFD, B_USB_RAW_COMMAND_GET_CONFIGURATION_DESCRIPTOR_ETC, &command, sizeof(command)) ||
```

Comment:
This one is indeed a typo in the patch sent upstream and should be fixed there as well.